tests/model/DataExtensionTest.php

```diff
@@ -159,13 +159,20 @@ public function testExtensionCanBeAppliedToDataObject() {
 		$do = new DataObject();
 		$mo = new DataExtensionTest_MyObject();
 
-		$this->assertTrue($do->hasMethod('testMethodApplied'));
 		$this->assertTrue($mo->hasMethod('testMethodApplied'));
+		$this->assertTrue($do->hasMethod('testMethodApplied'));
 
 		$this->assertEquals("hello world", $mo->testMethodApplied());
 		$this->assertEquals("hello world", $do->testMethodApplied());
 	}
 
+	public function testExtensionAllMethodNamesHasOwner() {
+		/** @var DataExtensionTest_MyObject $do */
+		$do = DataExtensionTest_MyObject::create();
```

Comment:
In the tests, it's a bit a mix between `new` and `::create()`. I just went with create because it's what I use most.